src/main/java/ru/mycubecraft/engine/graph/CollisionPrediction.java

```diff
@@ -0,0 +1,78 @@
+package ru.mycubecraft.engine.graph;
+
+import ru.mycubecraft.data.Contact;
+import ru.mycubecraft.world.BlockField;
+
+import java.util.List;
+
+import static java.lang.Float.NEGATIVE_INFINITY;
+import static java.lang.Float.POSITIVE_INFINITY;
+import static java.lang.Math.*;
+import static java.lang.Math.min;
+import static ru.mycubecraft.world.player.Player.*;
+import static ru.mycubecraft.world.player.Player.PLAYER_WIDTH;
+
+/**
+ * author chanpreet singh
+ * This class is built to predict the collison between two players and restrict them at a given coordinate
+ */
+
+public class CollisionPrediction {
+
+
+
+    /**
+     * Compute the exact collision point between the player and the block at <code>(x_coord, y_coord, z_coord)</code>.
+     * https://www.gamedev.net/tutorials/programming/general-and-gameplay-programming/swept-aabb-collision-detection-and-response-r3084/
+     */
+    public static void intersectSweptAabbAabb(int x_coord, int y_coord, int z_coord, float px, float py, float pz, float dx, float dy, float dz,
+                                        List<Contact> contacts, BlockField blockField) {
+        float pxmax = px + PLAYER_WIDTH, pxmin = px - PLAYER_WIDTH, pymax = py + PLAYER_HEIGHT - PLAYER_EYE_HEIGHT, pymin = py - PLAYER_EYE_HEIGHT,
+                pzmax = pz + PLAYER_WIDTH, pzmin = pz - PLAYER_WIDTH;
+
+        float xInvEntry = dx > 0f ? -pxmax : 1 - pxmin,
+                xInvExit = dx > 0f ? 1 - pxmin : -pxmax;
+
+        //boolean xNotValid = dx == 0 || blockField.load(x_coord + (dx > 0 ? -1 : 1), y_coord, z_coord) != null;
+        boolean xNotValid = dx == 0;
```

Comment:
There are in Java boolean naming convention, with prefix `is/has`?

`boolean isXNotValid = dx == 0`